### DISCUSSION TITLE
blade.php files are detected as `php` instead of `blade`

### DISCUSSION THREAD

**Discussion by Tmeister** [LazyVim/LazyVim#2639] [Q&A] [answered]
Hi,

I'm trying to set up tree-sitter, linting and formatting for blade files (`*.blade.php`). I was able to find the `blade-formatter` on Mason. 

The issue I'm seeing is that the (`*.blade.php`) files are detected as `php` files (see image one) and the formater is set to `pint` the PHP one (see image two).

I try to force the blade detection by adding the following to my **ftdetect** folder.

```lua
vim.filetype.add({
  extension = {
    astro = "astro",
    [".*%.blade%.php"] = "blade",
  },
})
```
The **astro** option works great, but the blade one doesn't. 

Can anyone point me in the right direction here?

Thanks!

![CleanShot 2024-03-01 at 15 53 59@2x](https://github.com/LazyVim/LazyVim/assets/210349/7c0240c8-d450-42d7-ab2f-21c2f8bec651)

![CleanShot 2024-03-01 at 15 55 09@2x](https://github.com/LazyVim/LazyVim/assets/210349/2c726f89-788d-431e-b067-0df1fb9d3b83)

**Comment by dpetka2001** [comment, marked as answer]
That's because you use the `extension` attribute of `vim.filetype.add`. You should use the `pattern` attribute like so
```lua
vim.filetype.add({
  pattern = {
    [".*%.blade%.php"] = "blade",
  },
})
```

**Reply by Tmeister** [reply]
Thank you!

That was it.

It's working now!
